packages/teams-js/src/public/dialog.ts

```diff
@@ -122,13 +133,22 @@ export namespace dialog {
   export function sendMessageToParentFromDialog(
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     message: any,
-  ): void {
+  ): Promise<void> {
     ensureInitialized(FrameContexts.task);
     if (!isSupported()) {
       throw errorNotSupportedOnPlatform;
     }
 
-    sendMessageToParent('messageForParent', [message]);
+    // if actionInfo exists in appContext, the dialog was loaded from a 3rd party app,
+    // and if sendMessageToParentFromDialog was called with a message,
+    // throw an error because the message cannot be passed along to the desired recipient
+    return app.getContext().then((appContext) => {
+      if (message && appContext.actionInfo) {
```

Comment:
If I am understanding it correctly, we would not want to allow `sendMessageToParentFromDialog` at all if `actionInfo` is provided. No need to check for message

Reply:
you are correct